app/helpers/lookup_keys_helper.rb

```diff
@@ -68,53 +68,123 @@ def validator_type_selector(f)
                "</dl>"), :title => _("Validation types")).html_safe}
   end
 
-  def overridable_lookup_keys(klass, host)
-    klass.class_params.override.where(:environment_classes => {:environment_id => host.environment}) + klass.lookup_keys
+  def overridable_lookup_keys(klass, host_or_hostgroup)
+    klass.class_params.override.where(:environment_classes => {:environment_id => host_or_hostgroup.environment}) + klass.lookup_keys
   end
 
-  def key_with_diagnostic(obj, key, index, lookup_value, value_hash = nil)
-    if obj.class.model_name == "Hostgroup"
-      value, matcher = obj.inherited_lookup_value key
-    else # host
-      value_for_key = value_hash[key.id] && value_hash[key.id][key.key]
-      value, matcher = value_for_key ? [value_for_key[:value], "#{value_for_key[:element]} (#{value_for_key[:element_name]})"] : [key.default_value, _("Default value")]
+  def can_edit_params?
+    authorized_via_my_scope("host_editing", "edit_params")
+  end
+
+  def lookup_key_with_diagnostic(obj, lookup_key, lookup_value)
+    value, matcher = value_matcher(obj, lookup_key)
+    inherited_value = lookup_key.value_before_type_cast(value)
+    overridden  = inherited_value.present? || lookup_value.value.present?
+    explanation = lookup_key_explanation(lookup_key, matcher, inherited_value)
```

Comment:
Maybe a comment here explaining what's `explanation` and what's `diagnostic` is needed, it wasn't obvious to me at first